client/extensions/woocommerce/state/sites/shipping-zones/actions.js

```diff
@@ -34,19 +35,22 @@ export const fetchShippingZones = ( siteId ) => ( dispatch, getState ) => {
 
 	dispatch( getAction );
 
-	return request( siteId ).get( 'shipping/zones' )
-		.catch( err => {
-			dispatch( setError( siteId, getAction, err ) );
-		} )
-		.then( ( data ) => {
-			if ( ! data ) {
-				return;
-			}
-			dispatch( fetchShippingZonesSuccess( siteId, data ) );
-			return Promise.all( data.map( zone => {
-				return fetchShippingZoneMethods( siteId, zone.id )( dispatch, getState );
-			} ).concat( data.map( zone => {
-				return fetchShippingZoneLocations( siteId, zone.id )( dispatch, getState );
-			} ) ) );
-		} );
+	return Promise.all( [
+		request( siteId ).get( 'shipping/zones' ),
+		fetchShippingMethods( siteId )( dispatch, getState ),
+	] )
+	.catch( err => {
+		dispatch( setError( siteId, getAction, err ) );
+	} )
+	.then( ( [ zoneData ] ) => {
```

Comment:
If fetching fails, the `.then()` function will be called without parameters, so you can't use array destructuring here.